[AlphaPyke] iP

AlphaPyke opened the pull request:
# Broski
> Your trusty gymbro-t

Broski is essentially your biological brother. He's got your back and he's here to:
- tell you your list as and when you ask
- keep track of your tasks

All you need to do is:
1. download the JAR file [here](https://github.com/AlphaPyke/ip/releases/download/A-Jar/Broski.jar)
2. java -jar it in terminal
3. add your tasks
4. let it do all the talking and management! :+1:

Best of all, it's completely **FREE** bro!

Features:
- [x] Managing tasks
- [ ] Reminders

If you wanna take a look at a snippet of the code:
```java
public static void main(String[] args) {
    Broski bot = new Broski();
    bot.start();
    bot.run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Rename, Greet, Exit

Commit:
Echo

Commit:
Add, List

Commit:
Mark as Done

Commit:
Todos, Events, Deadlines

Commit:
Automated Text UI Testing

Commit:
Handle Errors

Commit:
Delete

Commit:
Add save and load functionality

Commit:
Add date and time structure

To make the date and time display clearer, made it readable for users so that they can understand better

Commit:
Merge back to master branch

Commit:
Change file name for storage to match bot name

Commit:
Change name of path to match text file

Commit:
Do more OOP to ensure better OOP design

Commit:
Merge back to master branch

Commit:
Fix some coding standard

Commit:
Move all classes to broski package

Commit:
Merge to move all classes to broski package

Commit:
Merge branch 'add-gradle-support'

Commit:
Add tests for Todo and Broski class

Commit:
Modify build.gradle

Commit:
Add Javadocs for all existing methods

Commit:
Add find functionality

Commit:
Merge JavaDoc branch with master

Commit:
Merge conflict

Commit:
Add find functionality

Commit:
Add getter for task class

Commit:
Add GUI for Broski

Commit:
Add GUI for Broski and package classes appropriately

Commit:
Merge Level-10 back to master branch

mingyang143 commented on src/main/java/broski/output/Broski.java:
```diff
@@ -0,0 +1,107 @@
+
+import java.util.Scanner;
+
+import broski.exception.DeadlineException;
```
I like that the imports are very specific and not done with *. Well done!

mingyang143 commented on src/main/java/broski/taskRelated/TaskManager.java:
```diff
@@ -0,0 +1,110 @@
+     * @param line each line of the txt file.
+     * @return the specific task indicated by that line in the txt file.
+     */
+    private Task parseTask(String line) {
```
I like that the methods are named very well as it explains to the reader what it does accurately and at a sufficient level of detail.

mingyang143 commented on src/main/java/broski/taskRelated/TaskManager.java:
```diff
@@ -0,0 +1,110 @@
+ */
+public class TaskManager {
+
+    private static final String FILE_PATH = "./data/broski.txt";
```
I like that all constants are named correctly. Excellent!

kahsuann replied:
Agreed! It's great that you used SCREAMING_SNAKE_CASE for the constant

mingyang143 commented on src/test/java/broski/BroskiTest.java:
```diff
@@ -0,0 +1,24 @@
+
+public class BroskiTest {
+    @Test
+    public void start_checkGreeting_success(){
```
I like that the naming of the test methods are according to the Java coding standard.

mingyang143 commented on src/main/java/broski/task/Task.java:
```diff
@@ -0,0 +1,58 @@
+ */
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
```
I like that the boolean variable is made to sound like booleans.

mingyang143 reviewed (commented):
LGTM. Keep it up!

kahsuann commented on src/main/java/broski/output/Ui.java:
```diff
@@ -0,0 +1,227 @@
+        printLine();
+    }
+
+    /**
+     * Informs the user that the task has been deleted.
+     * @param taskList stores all prior task input.
+     * @param parser parse user input.
+     * @param reply user input.
+     */
```
I like that your Javadoc comments are very informative and have the correct form!

kahsuann commented on src/main/java/broski/task/Task.java:
```diff
@@ -0,0 +1,58 @@
+        this.isDone = false;
+    }
+
+    public String getDescription() {
```
It's great that your method names are verbs and written in camelCase!

kahsuann commented on src/main/java/broski/taskRelated/TaskManager.java:
```diff
@@ -0,0 +1,110 @@
+     * Saves task to hard disk as txt file.
+     * @param tasks list of tasks to be saved to hard disk.
+     */
+    public void saveTasks(List<Task> tasks) {
```
I like that you used plural form to represent a collection of tasks clearly!

kahsuann reviewed (commented):
LGTM! Indentations and line lengths are appropriate, and logical units are well-separated, making your code easily readable.

Commit:
Finish GUI for Broski

Commit:
Merge back to master

Commit:
Make minor changes to remove unnecessary lines

Commit:
Add an assert statement

To ensure that the variable at that point does not contain these segment, I added that assert statement.

Commit:
Check code quality and check style at the same time

AlphaPyke committed:
Merge pull request #1 from AlphaPyke/branch-A-Assertions

Add an assert statement

AlphaPyke committed:
Merge pull request #2 from AlphaPyke/branch-A-CodeQuality

Check code quality and check style at the same time

Commit:
Set up Continuous Integration

Commit:
Experiment with CI

Commit:
Return to original

Commit:
Add archive functionality

For now, archiving and listing of archive can be done.

Delete function not added for users however.

Commit:
Add product screenshot

Commit:
Make user guide

Commit:
Change issue in README

Commit:
Fix task saving issue

Commit:
Fix save task functionality again
